src/aliceVision/camera/Equidistant.hpp

```diff
@@ -9,175 +9,419 @@
+
 #include <vector>
 #include <sstream>
+#include <cmath>
 
 
 namespace aliceVision {
 namespace camera {
 
-/// Define an equidistant camera (store a K 3x3 matrix)
-///  with intrinsic parameters defining the K calibration matrix
-class Equidistant : public IntrinsicBase
+/**
+ * @brief EquiDistant is a camera model used for fisheye optics.
+ * See https://en.wikipedia.org/wiki/Fisheye_lens
+ * 
+ */
+class EquiDistant : public IntrinsicsScaleOffsetDisto
 {
-  public:
+public:
+  EquiDistant() = default;
+
+  EquiDistant(unsigned int w, unsigned int h, double fov, double ppx, double ppy,
+              std::shared_ptr<Distortion> distortion = nullptr)
+      : IntrinsicsScaleOffsetDisto(w, h, fov, fov, ppx, ppy, distortion)
+      , _circleRadius(std::min(w, h) * 0.5)
+      , _circleCenter(w / 2.0, h / 2.0)
+  {
+  }
 
-  Equidistant() = default;
+  EquiDistant(unsigned int w, unsigned int h, double fov, double ppx, double ppy, double circleRadiusPix,
+              std::shared_ptr<Distortion> distortion = nullptr)
+      : IntrinsicsScaleOffsetDisto(w, h, fov, fov, ppx, ppy, distortion)
+      , _circleRadius(circleRadiusPix)
+      , _circleCenter(w / 2.0, h / 2.0)
+  {
+  }
+
+  ~EquiDistant() override = default;
 
-  Equidistant(
-    unsigned int w, unsigned int h,
-    const Mat3 K)
-    :IntrinsicBase(w,h)
+  EquiDistant* clone() const override
   {
-    _K = K;
-    _Kinv = _K.inverse();
+    return new EquiDistant(*this); 
   }
 
-  Equidistant(
-    unsigned int w, unsigned int h,
-    double focal_length_pix,
-    double ppx, double ppy, const std::vector<double>& distortionParams = {})
-    : IntrinsicBase(w,h)
-    , _distortionParams(distortionParams)
+  void assign(const IntrinsicBase& other) override
   {
-    setK(focal_length_pix, ppx, ppy);
+    *this = dynamic_cast<const EquiDistant&>(other); 
   }
 
-  virtual ~Equidistant() {}
+  bool isValid() const override
+  {
+    return _scale(0) > 0 && IntrinsicBase::isValid(); 
+  }
 
-  virtual Equidistant* clone() const override { return new Equidistant(*this); }
-  virtual void assign(const IntrinsicBase& other) override { *this = dynamic_cast<const Equidistant&>(other); }
-  
-  virtual bool isValid() const override { return focal() > 0 && IntrinsicBase::isValid(); }
-  
-  virtual EINTRINSIC getType() const override { return EQUIDISTANT_CAMERA; }
-  std::string getTypeStr() const { return EINTRINSIC_enumToString(getType()); }
+  EINTRINSIC getType() const override
+  {
+    return EQUIDISTANT_CAMERA; 
+  }
 
-  double getFocalLengthPix() const { return _K(0,0); }
+  Vec2 project(const geometry::Pose3& pose, const Vec3& pt, bool applyDistortion = true) const override
+  {    
+    double rsensor = std::min(sensorWidth(), sensorHeight());
+    double rscale = sensorWidth() / std::max(w(), h());
+    double fmm = _scale(0) * rscale;
+    double fov = rsensor / fmm;
 
-  Vec2 getPrincipalPoint() const { return Vec2(_K(0,2), _K(1,2)); }
+    const Vec3 X = pose.rotation() * pt;
```

Comment:
Shouldn't the point be brought in the camera reference system? Why only applying the rotation and not the translation?
Am I missing something?